internal/pkg/deploy/cloudformation/app.go

```diff
@@ -71,12 +70,12 @@ func (cf CloudFormation) UpgradeApplication(in *deploy.CreateAppInput) error {
 	}
 	in.DNSDelegationAccounts = stack.DNSDelegatedAccountsForStack(appStack.SDK())
 	appConfig = stack.NewAppStackConfig(in)
-	s, err := toStack(appConfig)
-	if err != nil {
-		return err
-	}
-	if err := cf.upgradeAppStack(s); err != nil {
-		return err
+	if err := cf.upgradeAppStack(appConfig); err != nil {
+		var empty *cloudformation.ErrChangeSetEmpty
+		if !errors.As(err, &empty) {
+			return err
+		}
+		log.Infof("No changes for %s.\n", appConfig.StackName())
```

Comment:
We don't usually log outside of the `cli` pkg. Can we make it an error and catch it in the `cli` pkg to log it?

Reply:
hmmm...i don't think there's a good way for us to break that up without inserting some knowledge about the two stacks for the app into `cli` - essentially moving the two steps (`UpgradeAppStack` and `UpgradeAppStackSet`) over to `cli`. we could:
a) remove the log
b) break `UpgradeApplication` into `UpgradeAppStack()` and `UpgradeAppStackSet()`, and call those directly from `cli`
c) leave as is.

i'm happy with any of these options - thoughts?

Reply:
Why not a), does it look awkward?

Reply:
discussed offline - removing the log message!